neps/nep-0393.md

```diff
@@ -0,0 +1,542 @@
+
+### SBT Registry
+
+Atomicity of _soul transfer_ in current NEAR runtime is not possible if the token balance is kept separately for each SBT smart contract. We need an additional contract: the `SBT Registry`, to provide atomic transfer of all user SBTs and efficient way to ban accounts in relation to the Ban event. The registry provides a balance book of all associated SBT tokens. So, each SBT is requested by a smart contract (issuer), however it's is created in the registry.
+
+We can have multiple competing registries (with different purpose or different management scheme). Registries can utilize the same SBT contracts. An SBT smart contract, SHOULD opt-in to a registry. Registries may develop different opt-in mechanisms (they could differ by the approval mechanism, be fully permissioned etc..). One SBT smart contract can opt-in to:
+
+- many registries: it MUST relay all state change functions to all registries.
+- or to no registry: it MUST issue the SBT state change emits by itself.
+
+If an SBT smart contract doesn't opt-in to any registry, then we should think about it as a single token registry, and it MUST strictly implement all SBT Registry query functions by itself: the contract address must be part of the arguments, and it must check that it equals to the deployed account address (`require!(ctr == env::current_account_id())`).
+
+Moreover, a registry provides an efficient way to query multiple tokens for a single user. This will allow implementation of use cases such us:
+
+- SBT based identities (main use case of the `i-am-human` protocol);
+- SBT classes;
+- decentralized societies.
+
+#### Recovery within an SBT Registry
+
+SBT registry can define it's own mechanism to atomically recover all tokens related to one account and execute soul transfer to another account, without going one by one through each SBT issuer (sometimes that might be even not possible). Example mechanisms a Registry can implement to recovery mechanism:
+
+- KYC based recovery
+- Social recovery
+
+![Social Recovery, Image via “Decentralized Society”](https://bankless.ghost.io/content/images/public/images/cdb1fc23-6179-44f0-9bfe-e5e5831492f7_1399x680.png)
+
+SBT Registry based recovery is not part of this specification.
+
+#### Minting authorization
+
+A registry can limit which contracts can mint SBTs by implementing a custom issuer registration methods. Example: a simple access control list managed by a DAO.
+
+#### Burning tokens
+
+Registry MAY expose a mechanism to allow an account to burn an unwanted token. The exact mechanism is not part of the standard and it will depend on the registry implementation. We only define a standard `Burn` event, which must be emitted each time a token is removed from existence. Some registries may not allow to let accounts burning their tokens in a spirit of preserving specific claims. Ultimately, NEAR is a public blockchain, and even if a token is burned, it's trace will be preserved.
```

Comment:
```suggestion
Registry MAY expose a mechanism to allow an account to burn an unwanted token. The exact mechanism is not part of the standard and it will depend on the registry implementation. We only define a standard `Burn` event, which must be emitted each time a token is removed from existence. Some registries may not allow to let accounts burn their tokens in a spirit of preserving specific claims. Ultimately, NEAR is a public blockchain, and even if a token is burned, it's trace will be preserved.
```